src/Reflection/DataTransferObjectClass.php

```diff
@@ -66,6 +67,22 @@ public function validate(): void
 
     public function isStrict(): bool
     {
-        return $this->isStrict ??= ! empty($this->reflectionClass->getAttributes(Strict::class));
+        return $this->isStrict ??= ! empty($this->getStrictAttributesWithParent($this->reflectionClass));
+    }
+
+    /**
+     * @param array<ReflectionAttribute> $attributes
+     *
+     * @return array<ReflectionAttribute>
+     */
+    private function getStrictAttributesWithParent(ReflectionClass $reflectionClass, array $attributes = []): array
+    {
+        $attributes = array_merge($attributes, $reflectionClass->getAttributes(Strict::class));
+        $parent = $reflectionClass->getParentClass();
```

Comment:
This should be in a loop because there could several levels of inheritance. I've done something like this in the past:

```php
$attribute = null;

while ($attribute === null && $reflectionClass) {
    $attribute = $reflectionClass->getAttributes(Strict::class)[0] ?? null;

    $reflectionClass = $reflectionClass->getParent();
}
```

Reply:
I haven't tried this out, but it's something along those lines.

Reply:
I have used recursion which does the same, see passing test. Do you prefer a loop?

Reply:
Yes, a loop is clearer than recursive functions.